mbuild/recipes/monolayer.py

```diff
@@ -41,8 +42,30 @@ def __init__(self, surface, chain, backfill=None, pattern=None, tile_x=1,
         if pattern is None:  # Fill the surface.
             pattern = mb.Random2DPattern(len(tiled_compound.referenced_ports()))
 
-        # Attach chains to specified binding sites. Remaining sites get a backfill.
-        chains, backfills = pattern.apply_to_compound(guest=chain,
-                host=self['tiled_surface'], backfill=backfill, **kwargs)
-        self.add(chains)
+        if isinstance(chains, mb.Compound):
+            chains = [chains]
+
+        n_chains = len(pattern.points)
+
+        # Attach chains of each type to binding sites based on specified fractions
+        for i,chain in enumerate(chains[:-1]):
+
+            # Create sub-pattern for this chain type
+            subpattern = deepcopy(pattern)
+            n_points = round(fractions[i] * n_chains)
+            points = subpattern.points[np.random.choice(subpattern.points.shape[0], n_points, replace=False)]
+            subpattern.points = points
+
+            # Remove now-occupied points from overall pattern
+            pattern.points = np.array([point for point in pattern.points.tolist() if point not in subpattern.points.tolist()])
+
+            # Attach chains to the surface
+            attached_chains = subpattern.apply_to_compound(guest=chain,
```

Comment:
This returns a tuple. Just to be explicit, I would write `attached_chains, _ = subpat...`

I think it ultimately doesn't matter because `Compound.add()` will go in and only pick out compounds in the tuple of lists but just to make it slightly less confusing when reading.